### PR TITLE
Add rfc for feature 'placement rules in SQL works with tables with TiFlash replica'

### LINKED ISSUE #5768: Add RFC for feature 'placement rules in SQL works with tables with TiFlash replica'
## Enhancement

Add RFC for feature 'placement rules in SQL works with tables with TiFlash replica'.

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: close #5768 ref https://github.com/pingcap/tidb/issues/37171

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: docs/design/2022-08-31-placement-rules-in-SQL-works-with-tables-with-TiFlash-replica.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+The conflict between tiflash replicas and placement rules setting is because both setting tiflash replicas and setting placement rules will generate placement configuration in PD. In the configuration generated by setting placement rules, ```group_index = 40, group_override = true```; while in the configuration generated by tiflash replica, ```group_index = 0```, ```group_override = false```. According to the rules in PD, if ```group_override``` is true , the configuration will override the configuration of other groups whose ```group_index``` is smaller than its own; if ```group_override``` is false, other configurations will not be overridden. Therefore, placement rules will overwrite the setting of tiflash replicas. In order to avoid this problem, the coexistence of placement rules and tiflash replicas was prohibited in the original development.
```
Comment (breezewish) [inline docs/design/2022-08-31-placement-rules-in-SQL-works-with-tables-with-TiFlash-replica.md:13]: Usually we use single backquote for inlined codes. The triple backquote is used for block level codes. See markdown syntax: https://www.freecodecamp.org/news/how-to-format-code-in-markdown/

File: docs/design/2022-08-31-placement-rules-in-SQL-works-with-tables-with-TiFlash-replica.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Placement rules in SQL works with tables with TiFlash replica
```
Comment (breezewish) [inline docs/design/2022-08-31-placement-rules-in-SQL-works-with-tables-with-TiFlash-replica.md:1]: Currently we do not have a linter rule in CI for markdowns. Prefer to install the "Prettier" extension in VsCode and use it to format your markdown.

**Review by flowbehappy — APPROVED** [review]

**Pull request merged**